ch_tools/chadmin/internal/table.py

```diff
@@ -450,14 +399,25 @@ def delete_detached_table(ctx, database_name, table_name):
             f"No metadata file for table '{escaped_database_name}'.'{escaped_table_name}' by path {local_metadata_table_path}."
         )
 
-    check_table_engine_supported(local_metadata_table_path)
+    table_metadata = parse_table_metadata(local_metadata_table_path)
 
-    table_uuid = _get_uuid_table(local_metadata_table_path)
     for disk_type, disk_name in _get_disks_data(ctx).items():
         _remove_table_data_from_disk(
-            table_uuid=table_uuid, disk_name=disk_name, disk_type=disk_type
+            table_uuid=table_metadata.table_uuid,
+            disk_name=disk_name,
+            disk_type=disk_type,
         )
 
+    if table_metadata.table_engine.is_table_engine_replicated():
+        current_node = table_metadata.replica_path + "/replicas/"
```

Comment:
I think better to use clean_zk_metadata_for_hosts, because it is duplicating same logic, also logic of removing host might be more complicated sometimes: we want to remove all replica name mentioning from the zookeeper, not only from /replicas/.